src/WebJobs.Extensions.MobileApps/Config/MobileAppsConfiguration.cs

```diff
@@ -53,8 +62,194 @@ public void Initialize(ExtensionConfigContext context)
                 throw new ArgumentNullException("context");
             }
 
-            context.Config.RegisterBindingExtension(
-                new MobileTableAttributeBindingProvider(context.Config, this, context.Config.NameResolver));
+            INameResolver nameResolver = context.Config.GetService<INameResolver>();
+            IConverterManager converterManager = context.Config.GetService<IConverterManager>();
+
+            BindingFactory factory = new BindingFactory(nameResolver, converterManager);
+
+            IBindingProvider outputProvider = factory.BindToGenericAsyncCollector<MobileTableAttribute, MobileTableContext>(typeof(MobileTableAsyncCollector<>), BindForOutput);
+
+            IBindingProvider clientProvider = factory.BindToExactType<MobileTableAttribute, IMobileServiceClient>(BindForClient);
+
+            IBindingProvider queryProvider = factory.BindToGenericItem<MobileTableAttribute>(BindForQuery);
+            queryProvider = factory.AddTypeFilter(IsQueryType, queryProvider);
+            queryProvider = factory.AddValidator<MobileTableAttribute>(ThrowIfGenericArgumentIsInvalid, queryProvider);
+
+            IBindingProvider jObjectTableProvider = factory.BindToExactType<MobileTableAttribute, IMobileServiceTable>(BindForTable);
+
+            IBindingProvider tableProvider = factory.BindToGenericItem<MobileTableAttribute>(BindForTable);
+            tableProvider = factory.AddTypeFilter(IsTableType, tableProvider);
+            tableProvider = factory.AddValidator<MobileTableAttribute>(ThrowIfGenericArgumentIsInvalid, tableProvider);
+
+            IBindingProvider itemProvider = factory.BindToGenericItem<MobileTableAttribute>(BindForItem);
+            itemProvider = factory.AddValidator<MobileTableAttribute>(IsItemType, itemProvider);
+
+            IExtensionRegistry extensions = context.Config.GetService<IExtensionRegistry>();
+            extensions.RegisterBindingRules<MobileTableAttribute>(ValidateMobileAppUri, nameResolver, outputProvider, clientProvider, jObjectTableProvider, queryProvider, tableProvider, itemProvider);
+        }
+
+        internal bool IsQueryType(Type paramType)
```

Comment:
I actually debating going further -- having a separate class per Binding (so one to setup ItemBinding, one to setup QueryBinding,one to set up OutputBinding etc). So I think I'd end up with 6 small classes from this, which is still a massive improvement over the classes I had before. 

To me, that makes it clearer where my functionality lies. And when I want to support something new, it's clear that I'd need to add a new class with the logic, and then wire it up with a line of code in the Config.

I've discussed that with Mike and I'll see what it looks like -- but not in this PR.

---
In reply to: [63897796](https://github.com/Azure/azure-webjobs-sdk-extensions/pull/97#discussion_r63897796) [](ancestors = 63897796)